blog-posts/en/angular-http-only-cookie.md

```diff
@@ -0,0 +1,423 @@
+    }
+ 
+    public async Task Invoke(HttpContext context)
+    {
+        if (context.Request.Cookies.TryGetValue("Abp.AuthToken", out var token))
+        {
+            context.Request.Headers.Append("Authorization", "Bearer " + token);
+        }
+
+        await _next.Invoke(context);
+    }
+}
+
+public static class HttpOnlyMiddlewareExtension
+{
+    public static IApplicationBuilder UseHttpOnlyToken(this IApplicationBuilder builder)
+    {
+        return builder.UseMiddleware<HttpOnlyMiddleware>();
+    }
+}
+```
+
+### Configure `Startup.cs`
+
+We need to add the following code to `Startup.cs` file in your `*.Web.Host` project.
+
+```csharp
+// use before Authentication middleware
+app.UseHttpOnlyToken();
+
+app.UseAuthentication();
+```
+
+## Step 3: Configure your `Angular` project
+
+You need to remove cookies from your Angular application and a workaround for tenant switching.
+
+### Remove cookie methods
+
+We need to update `abp.js` file in your `src/assets/abp-web-resources` folder. 
+
+Find the `abp.auth.setToken` and `abp.auth.getToken` functions, then comment this functions.
+
+```js
+// abp.auth.setToken = function (authToken, expireDate) { abp.utils.setCookieValue(abp.auth.tokenCookieName, authToken, expireDate, abp.appPath, abp.domain, { 'SameSite': abp.auth.tokenCookieSameSite });
+};
+```
+
+Let's delete the places where the `abp.auth.setToken` function is used
+
+* src/AppPreBootstrap.ts
+* src/account/auth/zero-refresh-token.service.ts
+* src/account/login/login.service.ts
+
+Let's update the places where the `abp.auth.getToken` function is used
+
+```js
+private static getUserConfiguration(callback: () => void): any {
+    return XmlHttpRequestHelper.ajax(
+        'GET',
+        AppConsts.remoteServiceBaseUrl + '/AbpUserConfiguration/GetAll',
+        null,
+        null,
+        (response) => {
+            let result = response.result;
+
+            _merge(abp, result);
+
+            abp.clock.provider = this.getCurrentClockProvider(result.clock.provider);
+
+            AppPreBootstrap.configureLuxon();
+
+            abp.event.trigger('abp.dynamicScriptsInitialized');
+
+            AppConsts.recaptchaSiteKey = abp.setting.get('Recaptcha.SiteKey');
+            AppConsts.subscriptionExpireNootifyDayCount = parseInt(
+                abp.setting.get('App.TenantManagement.SubscriptionExpireNotifyDayCount')
+            );
+
+            DynamicResourcesHelper.loadResources(callback);
+        }
+    );
+}
+```
+
+Update `src/app/shared/common/auth/app-auth.service.ts` file as below:
+
+```ts
+@Injectable()
+export class AppAuthService {
+    logout(reload?: boolean, returnUrl?: string): void {
+        XmlHttpRequestHelper.ajax(
+            'GET',
+            AppConsts.remoteServiceBaseUrl + '/api/TokenAuth/LogOut',
+            null,
+            null,
+            () => {
+                this.logoutInternal(reload, returnUrl);
+            },
+            (errorResult: IAjaxResponse) => {
```

Comment:
I'm getting a syntax error here because `XmlHttpRequestHelper` expects only 5 arguments, but there are 6. Not sure which of the logout functions should be used.

Reply:
Hi @StarbuckSapien. There are 6 arguments for this method. Could you check this again? https://github.com/aspnetzero/aspnet-zero-core/blob/9c4358d4a87f740a563fe78079d388c287190ad3/angular/src/shared/helpers/XmlHttpRequestHelper.ts#L2

Reply:
Ah, yes. It looks like that changed 4 months ago and my code is older than that. I'll just drop off the new last argument.